### PR TITLE
feat: converting general vis to Mantine and add hexbin visualization

### PR DESCRIPTION
**Pull request by dvzacharycutler**
Closes https://github.com/datavisyn/reprovisyn/issues/514

### Developer Checklist (Definition of Done)

**Issue**

- [x] All acceptance criteria from the issue are met
- [ ] Tested in latest Chrome/Firefox

**UI/UX/Vis**

- [x] Requires UI/UX/Vis review
  - [ ] Reviewer(s) are notified (_tag assignees_)
  - [ ] Review has occurred (_link to notes_)
  - [ ] Feedback is included in this PR
  - [ ] Reviewer(s) approve of concept and design

**Code**

- [x] Branch is up-to-date with the branch to be merged with, i.e., develop
- [x] Code is cleaned up and formatted
- [ ] Unit tests are written (frontend/backend if applicable)
- [ ] Integration tests are written (if applicable)

**PR**

- [x] Descriptive title for this pull request is provided (will be used for release notes later)
- [x] Reviewer and assignees are defined
- [ ] Add type label (e.g., *bug*, *feature*) to this pull request
- [ ] Add release label (e.g., `release: minor`) to this PR following [semver](https://semver.org/)
- [x] The PR is connected to the corresponding issue (via `Closes #...`)
- [x] [Summary of changes](#summary-of-changes) is written

### Summary of changes

- Mostly converting all UI components inside of Vis to mantine. Large changes to appearance, including making the sidebar slide out and cover part of the vis when opened. Also made the following small changes from https://github.com/datavisyn/reprovisyn/issues/514

-  Numerical Columns -> Sentence vs. title case
-  Scatterplot: axis starts from 0
-  Remove crosshair mouse pointer -> use regular mouse pointer!
-  Turn-off dashed lines on hover
-  Remove warning in vis side panel

### Screenshots

[finalMantineVis.webm](https://user-images.githubusercontent.com/87644761/191498093-780d4224-b0ee-4d77-bc32-db139fa010bc.webm)

### Additional notes for the reviewer(s)

-  
Thanks for creating this pull request 🤗

### CODE REVIEW

File: src/vis/sidebar/OpacitySlider.tsx [hunk 1]
```diff
@@ -1,26 +1,43 @@
...
-        max="1"
-        step=".1"
-        id="alphaSlider"
-      />
-    </div>
+      </Text>
+      <Box sx={{ width: '200px' }}>
+        <Slider
+          step={0.05}
+          value={currentValue}
```
Comment (thinkh) [inline src/vis/sidebar/OpacitySlider.tsx:28]: Please round values to two decimal digits after the decimal point (e.g., `0.30` or even better in this case `0.3`).

grafik

File: src/vis/sidebar/BrushOptionButtons.tsx [hunk 2]
```diff
@@ -8,52 +11,15 @@ interface BrushOptionProps {
...
-      </label>
-    </div>
+    <SegmentedControl
+      value={dragMode}
+      onChange={callback}
+      data={[
+        { label: <FontAwesomeIcon icon={faSquare} />, value: EScatterSelectSettings.RECTANGLE },
+        { label: <FontAwesomeIcon icon={faPaintBrush} />, value: EScatterSelectSettings.LASSO },
+        { label: <FontAwesomeIcon icon={faArrowsAlt} />, value: EScatterSelectSettings.PAN },
+        { label: <FontAwesomeIcon icon={faSearchPlus} />, value: EScatterSelectSettings.ZOOM },
```
Comment (thinkh) [inline src/vis/sidebar/BrushOptionButtons.tsx:21]: Please add some Mantine tooltips to the icons. It's currently unclear to the user what they mean.

grafik

File: src/vis/scatter/ScatterVis.tsx [hunk 3]
```diff
@@ -136,12 +131,16 @@ export function ScatterVis({
...
-    <div ref={plotlyDivRef} className="d-flex flex-row w-100 h-100" style={{ minHeight: '0px' }}>
-      <div
-        className={`position-relative d-flex justify-content-center align-items-center flex-grow-1 ${
-          traceStatus === 'pending' ? 'tdp-busy-partial-overlay' : ''
-        }`}
-      >
+    <Container fluid sx={{ flexGrow: 1, height: '100%', overflow: 'hidden' }} ref={plotlyDivRef}>
+      <ActionIcon sx={{ position: 'absolute', top: '10px', right: '10px' }} onClick={() => setSidebarOpen(true)}>
+        <FontAwesomeIcon icon={faGear} />
+      </ActionIcon>
```
Comment (thinkh) [inline src/vis/scatter/ScatterVis.tsx:137]: Please add a Mantine tooltip:

> Switch visualization type and configure currently selected visualization

Apply this change to all other action icons of the visualization types

File: src/vis/sidebar/VisTypeSelect.tsx [hunk 4]
```diff
@@ -9,21 +9,18 @@ interface VisTypeSelectProps {
...
+      label="Visualization type"
+      // components={{Option: optionLayout}}
+      onChange={(e) => callback(e as ESupportedPlotlyVis)}
+      name="visTypes"
+      data={allVisTypes.map((t) => {
+        return {
+          value: t,
+          label: t,
+        };
+      })}
```
Comment (thinkh) [inline src/vis/sidebar/VisTypeSelect.tsx:22]: Switch visualization type to sentence case: 

grafik

**Review by thinkh — CHANGES_REQUESTED** [review]
Thanks for the fresh UI. It looks much nicer and usable now. I have some minor suggestions and two additional points:

1. Why did you remove the PCP visualization?
2. I cannot run storybook with `yarn run storybook` (in tdp_core standalone) and get the following error:

```
❯ yarn run storybook
info @storybook/react v6.5.12
info 
info => Loading presets
info Addon-docs: using MDX1
info => Using implicit CSS loaders
(node:1903) DeprecationWarning: Default PostCSS plugins are deprecated. When switching to '@storybook/addon-postcss',
you will need to add your own plugins, such as 'postcss-flexbugs-fixes' and 'autoprefixer'.

See https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#deprecated-default-postcss-plugins for details.
(Use `node --trace-deprecation ...` to show where the warning was created)
info => Using default Webpack4 setup
ERR! TypeError: Cannot read properties of undefined (reading 'data')
ERR!     at setData (/home/holger/repos/tdp_core/node_modules/webpack-virtual-modules/index.js:84:15)
ERR!     at CachedInputFileSystem.finalInputFileSystem._writeVirtualFile (/home/holger/repos/tdp_core/node_modules/webpack-virtual-modules/index.js:118:9)
ERR!     at VirtualModulesPlugin.writeModule (/home/holger/repos/tdp_core/node_modules/webpack-virtual-modules/index.js:64:24)
ERR!     at /home/holger/repos/tdp_core/node_modules/webpack-virtual-modules/index.js:165:14
ERR!     at Array.forEach (<anonymous>)
ERR!     at afterResolversHook (/home/holger/repos/tdp_core/node_modules/webpack-virtual-modules/index.js:164:40)
ERR!     at Hook.eval [as call] (eval at create (/home/holger/repos/tdp_core/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:7:1)
ERR!     at Hook.CALL_DELEGATE [as _call] (/home/holger/repos/tdp_core/node_modules/tapable/lib/Hook.js:14:14)
ERR!     at WebpackOptionsApply.process (/home/holger/repos/tdp_core/node_modules/webpack/lib/WebpackOptionsApply.js:695:33)
ERR!     at createCompiler (/home/holger/repos/tdp_core/node_modules/webpack/lib/webpack.js:80:28)
ERR!  TypeError: Cannot read properties of undefined (reading 'data')
ERR!     at setData (/home/holger/repos/tdp_core/node_modules/webpack-virtual-modules/index.js:84:15)
ERR!     at CachedInputFileSystem.finalInputFileSystem._writeVirtualFile (/home/holger/repos/tdp_core/node_modules/webpack-virtual-modules/index.js:118:9)
ERR!     at VirtualModulesPlugin.writeModule (/home/holger/repos/tdp_core/node_modules/webpack-virtual-modules/index.js:64:24)
ERR!     at /home/holger/repos/tdp_core/node_modules/webpack-virtual-modules/index.js:165:14
ERR!     at Array.forEach (<anonymous>)
ERR!     at afterResolversHook (/home/holger/repos/tdp_core/node_modules/webpack-virtual-modules/index.js:164:40)
ERR!     at Hook.eval [as call] (eval at create (/home/holger/repos/tdp_core/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:7:1)
ERR!     at Hook.CALL_DELEGATE [as _call] (/home/holger/repos/tdp_core/node_modules/tapable/lib/Hook.js:14:14)
ERR!     at WebpackOptionsApply.process (/home/holger/repos/tdp_core/node_modules/webpack/lib/WebpackOptionsApply.js:695:33)
ERR!     at createCompiler (/home/holger/repos/tdp_core/node_modules/webpack/lib/webpack.js:80:28)

WARN Broken build, fix the error above.
WARN You may need to refresh the browser.

info => Loading presets
✔ Would you like to send crash reports to Storybook? … no

attention => Storybook now collects completely anonymous telemetry regarding usage.
This information is used to shape Storybook's roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://storybook.js.org/telemetry
```

Any idea what is missing here?

**Comment by thinkh** [comment]
> Why did you remove the PCP visualization?

According to @dvzacharycutler PCP was removed because it does not work well with Plotly. It should replaced with a better implementation in the future.

File: src/vis/sidebar/OpacitySlider.tsx (hunk 1 above)
Comment (dvzacharycutler) [reply src/vis/sidebar/OpacitySlider.tsx:28]: done

File: src/vis/sidebar/BrushOptionButtons.tsx (hunk 2 above)
Comment (dvzacharycutler) [reply src/vis/sidebar/BrushOptionButtons.tsx:21]: done

File: src/vis/scatter/ScatterVis.tsx (hunk 3 above)
Comment (dvzacharycutler) [reply src/vis/scatter/ScatterVis.tsx:137]: done

File: src/vis/sidebar/VisTypeSelect.tsx (hunk 4 above)
Comment (dvzacharycutler) [reply src/vis/sidebar/VisTypeSelect.tsx:22]: done

**Review by thinkh — APPROVED** [review]
I reviewed this together with @puehringer in a call and we refactored the show/hide sidebar to configure the visualization. The new flags are `enableSidebar`, `showSidebar` and `setShowSidebar` which allow using the Vis in controlled or uncontrolled state.

**Pull request merged**